pygsti/algorithms/contract.py

```diff
@@ -243,7 +243,7 @@ def _contract_to_cp_direct(model, verbosity, tp_also=False, maxiter=100000, tol=
             #  print "DEBUG: orig evals == ",evals                #DEBUG
             assert(abs(sum(new_evals) - 1.0) < 1e-8)
 
-            new_Jmx = _np.dot(evecs, _np.dot(_np.diag(new_evals), _np.linalg.inv(evecs)))
+            new_Jmx = evecs @ _np.diag(new_evals) @ inv_evecs
```

Comment:
Minor request. When doing matrix multiplication with a diagonal matrix it is typically much faster to do element-wise multiplication with the vector of diagonal matrix entries. This will get broadcast into an O(n^2) rather than O(n^3) operation.